src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+            iterable = mongoCollectionComments.find(Filters.and(Filters.eq("post_id", postId), Filters.gt("_id", startAfterId.toString())));
+        }
+        iterable.limit(10);
+        iterable.into(commentModels);
+        return commentModels;
+    }
+
+    private static List<CommentModel> getByDate(String postId, String date){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        List<CommentModel> commentList = new ArrayList<>();
+        iterable = mongoCollectionComments.find(Filters.and(eq("dateString", date), Filters.eq("post_id", postId))).limit(10);
+        iterable.into(commentList);
+        return commentList;
+    }
+
+    public static Route updateComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        CommentModel commentModel;
+        //PermissionChecker permissionChecker;
+        String commentId = paramHelper.valueOf("_id");
+        String bodyText = paramHelper.valueOf("bodyText");
+        commentModel = mongoCollection.findOne(eq("_id", new ObjectId(commentId)));
+        if(sessionManager.getCurrentUserId().equals(commentModel.authorId)){
+            commentModel.setBodyText(bodyText);
+            mongoCollection.findOneAndReplace(eq("_id", new ObjectId(commentId)), commentModel);
+            return " ok ";
+        }
+        return " not ok ";
+    });
+
+    public static Route destroyComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        JacksonMongoCollection<UserModel> mongoCollectionUser =
+                new CollectionFactory<UserModel>(ServiceLocator.getService().dbService(), UserModel.class).getCollection();
+        DeleteResult result;
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        String user = sessionManager.getCurrentUserId();
+        UserModel userModel = mongoCollectionUser.findOne(Filters.eq("_id", new ObjectId(user)));
+/*        UserModel userModel = new UserModel();
+        userModel.setAdmin(false);*/
+        String commentId = paramHelper.valueOf("_id");
+        CommentModel commentModel = mongoCollection.findOne(eq("_id", new ObjectId(commentId)));
+        if(commentModel == null){
+            return "comment DNE";
+        }
+        if(userModel.isAdmin()){
```

Comment:
this only allows admins to delete comments - the idea is that admins, moderators, users can delete comments.

do not worry about permission checking for now - allow any user to delete/update a comment. Permission checking for whom can alter a comment is pretty tricky and I will add the permission checks in.